### PR TITLE
Oracle - Implementation of Batch Insertion

### PR DESCRIPTION
**Pull request by ralmsdeveloper**
**Please check if the PR fulfills these requirements**

- [x] The code builds and tests pass (verified by our automated build checks)
- [x] Commit messages follow this format
``` 

  Oracle Batch Insert Implementation

    Fixes #10011
```
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Code meets the expectations our engineering guidelines. https://github.com/aspnet/Home/wiki/Engineering-guidelines.

Please review the guidelines for CONTRIBUTING.md for more details.

### CODE REVIEW

**Comment by ralmsdeveloper** [comment]
Hello @AndriySvyryd, this is the #10011 fix that was closed by me.

@AndriySvyryd, I did all the tests and they were satisfactory, some adjustments were made to remove the existing hack, with your help I did the implementation with OracleModificationCommandBatch!

So I believe that all adjustments and implementations are in compliance!

```
------ Test started: Assembly: Microsoft.EntityFrameworkCore.Oracle.FunctionalTests.dll ------

5595 passed, 0 failed, 78 skipped (see 'Error List / Messages'), took 486,58 seconds (xUnit.net 2.3.0 build 3820).
```


c/ @ajcvickers

**Comment by AndriySvyryd** [comment]
@ralmsdeveloper There's still one test failing: 
>Test 'Microsoft.EntityFrameworkCore.Query.UdfDbFunctionOracleTests.Scalar_Function_Where_Not_Correlated' failed: Assert.Equal() Failure
Expected: 2
Actual:   1
	C:\enlistments\Universe\EntityFramework\samples\OracleProvider\test\OracleProvider.FunctionalTests\Query\UdfDbFunctionOracleTests.cs(303,0): at Microsoft.EntityFrameworkCore.Query.UdfDbFunctionOracleTests.Scalar_Function_Where_Not_Correlated()

**Comment by ralmsdeveloper** [comment]
@AndriySvyryd, I had changed to wait 1, for me!

So I changed my "CultureInfo" to test!

 Because I thought this was incorrect, then I realized that because of my region the date format was influencing the return of my test!

File: samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs [hunk 1]
```diff
@@ -1,23 +1,247 @@
...
+using Oracle.ManagedDataAccess.Client;
+using System;
+using System.Data;
+using System.Linq;
+using Xunit;
 
 namespace Microsoft.EntityFrameworkCore.Query
 {
-    public class FromSqlSprocQueryOracleTest : FromSqlSprocQueryTestBase<NorthwindQueryOracleFixture<NoopModelCustomizer>>
+    public class FromSqlSprocQueryOracleTest : NorthwindQueryOracleFixture<NoopModelCustomizer>
```
Comment (AndriySvyryd) [inline samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs]: Why did you change the base type?

**Comment by AndriySvyryd** [comment]
@ralmsdeveloper Could you fix that test then, so it works with any culture? I think changing `startDate` should be enough:
```C#
var startDate = DateTime.Parse("1/4/2000", CultureInfo.InvariantCulture);
```

File: samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs (hunk 1 above)
Comment (ralmsdeveloper) [reply samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs]: I changed to be able to pass parameters of cursors, to avoid the undesirable hacks that had!

Comment (ralmsdeveloper) [reply samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs]: But if you do not like it, I can use "override" in the methods.

I particularly liked it, even for simplicity

Comment (AndriySvyryd) [reply samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs]: I see. We prefer to not repeat code if possible, this allows to make changes faster.
You could add methods like these to the base and override them:
```C#
protected virtual object[] GetTenMostExpensiveProductsParameters()
    => new object[0];
protected virtual object[] GetCustomerOrderHistorySprocParameters()
    => new object[0];
```

Comment (ralmsdeveloper) [reply samples/OracleProvider/test/OracleProvider.FunctionalTests/Query/FromSqlSprocQueryOracleTest.cs]: I liked that, I did not want to implement it without your analysis!

**Comment by ralmsdeveloper** [comment]
Coincidentally I had done it in my tests!

**Comment by ralmsdeveloper** [comment]
@AndriySvyryd, See if this change is right for you!

**Comment by AndriySvyryd** [comment]
Looking good, now you can do the same to `AsyncFromSqlSprocQueryOracleTest`

**Comment by ralmsdeveloper** [comment]
Done @AndriySvyryd 

I followed your suggestion!

**Comment by ralmsdeveloper** [comment]
Yeah, sure I'll do it right now!

**Comment by ralmsdeveloper** [comment]
I will prefer to send a new PR where in a single comite will be all changes, I think better!
